packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-belongs-to-transformer.test.ts

```diff
@@ -472,3 +473,277 @@ test('Should not resolve to secondary index of connected model if the index is d
+      });
+    
+      const out = transformer.transform(inputSchema);
+      expect(out).toBeDefined();
+      const schema = parse(out.schema);
+      validateModelSchema(schema);
+      const connectionFieldName1 = 'taskTodoTodoid';
+      const connectionFieldName2 = 'taskTodoName';
+      //Type definition
+      const objType = schema.definitions.find((def: any) => def.name && def.name.value === 'Task') as any;
+      expect(objType).toBeDefined();
+      const relatedField1 = objType.fields.find((f: any) => f.name.value === connectionFieldName1);
+      expect(relatedField1).toBeDefined();
+      expect(relatedField1.type.kind).toBe(Kind.NON_NULL_TYPE);
+      expect(relatedField1.type.type.name.value).toBe('ID');
+      const relatedField2 = objType.fields.find((f: any) => f.name.value === connectionFieldName2);
+      expect(relatedField2).toBeDefined();
+      expect(relatedField2.type.kind).toBe(Kind.NON_NULL_TYPE);
+      expect(relatedField2.type.type.name.value).toBe('String');
+      //Create Input
+      const createInput = schema.definitions.find((def: any) => def.name && def.name.value === 'CreateTaskInput') as any;
+      expect(createInput).toBeDefined();
+      expect(createInput.fields.length).toEqual(5);
+      const createInputConnectedField1 = createInput.fields.find((f: any) => f.name.value === connectionFieldName1);
+      expect(createInputConnectedField1).toBeDefined();
+      expect(createInputConnectedField1.type.kind).toBe(Kind.NON_NULL_TYPE);
+      expect(createInputConnectedField1.type.type.name.value).toBe('ID');
+      const createInputConnectedField2 = createInput.fields.find((f: any) => f.name.value === connectionFieldName2);
+      expect(createInputConnectedField2).toBeDefined();
+      expect(createInputConnectedField2.type.kind).toBe(Kind.NON_NULL_TYPE);
+      expect(createInputConnectedField2.type.type.name.value).toBe('String');
+      //Update Input
+      const updateInput = schema.definitions.find((def: any) => def.name && def.name.value === 'UpdateTaskInput') as any;
+      expect(updateInput).toBeDefined();
+      expect(updateInput.fields.length).toEqual(5);
+      const updateInputConnectedField1 = updateInput.fields.find((f: any) => f.name.value === connectionFieldName1);
+      expect(updateInputConnectedField1).toBeDefined();
+      expect(updateInputConnectedField1.type.kind).toBe(Kind.NAMED_TYPE);
+      expect(updateInputConnectedField1.type.name.value).toBe('ID');
+      const updateInputConnectedField2 = updateInput.fields.find((f: any) => f.name.value === connectionFieldName2);
+      expect(updateInputConnectedField2).toBeDefined();
+      expect(updateInputConnectedField2.type.kind).toBe(Kind.NAMED_TYPE);
+      expect(updateInputConnectedField2.type.name.value).toBe('String');
+    });
+  });
+
+  describe('@belongsTo with @hasMany', () => {
+    test('Should generate nullable connection fields in type definition and create/update input when belongsTo field is nullable', () => {
```

Comment:
Is this test is repeated twice since I see the same schema and description [here](https://github.com/aws-amplify/amplify-category-api/pull/817/files#diff-acd0902518c587805b3f7217654188e3cc323f5dc9500f146db4f3673109b1f6R490)

Reply:
The first one is `hasOne` and this one is `hasMany`